service/src/main/java/skills/skillLoading/SubjectDataLoader.groovy

```diff
@@ -178,9 +184,26 @@ class SubjectDataLoader {
         return skillsAndPoints;
     }
 
-    private List<SkillsAndPoints> handleSkillExpirations(String projectId, String userId, List<SkillsAndPoints> skillsAndPoints) {
+    private List<SkillsAndPoints> handleAchievements(String projectId, String userId, List<SkillsAndPoints> skillsAndPoints) {
         if(projectId) {
             List<String> skillIds = collectSkillIds(skillsAndPoints)
+            def achievedSkills = achievedLevelRepository.getAchievedDateByUserIdAndProjectIdAndSkillBatch(userId, projectId, skillIds)
+            if (achievedSkills) {
+                skillsAndPoints.each { it ->
+                    def achievements = achievedSkills.findAll{skill -> skill.skillId == it.skillDef.skillId}
+                    if(achievements) {
+                        achievements?.sort { skill -> skill }
```

Comment:
not sure if the sort by skillId make sense since the achievements were seleted by skillId, was it meant to be date?